go/chat/boxer.go

```diff
@@ -202,16 +202,13 @@ func (b *Boxer) unboxV1(ctx context.Context, boxed chat1.MessageBoxed, encryptio
 	// compute the header hash
 	headerHash := b.hashV1(boxed.HeaderCiphertext.E)
 
+	// Whether the body is missing (deleted)
+	skipBodyVerification := (len(boxed.BodyCiphertext.E) == 0)
+
 	// decrypt body
 	// will remain empty if the body was deleted
 	var bodyVersioned chat1.BodyPlaintext
-	skipBodyVerification := false
-	if len(boxed.BodyCiphertext.E) == 0 {
-		if boxed.ServerHeader.SupersededBy == 0 {
-			return nil, NewPermanentUnboxingError(errors.New("empty body and not superseded in MessageBoxed"))
```

Comment:
We didn't actually keep up with this constraint.

If you do this
```
msg 1: TEXT
msg 2: EDIT msg 1
msg 3: DELETE msg 1
```

You get this
```
msg 1: TEXT deleted, supersededBy 3
msg 2: EDIT deleted, supersededBy 0 <-- oops
msg 3: DELETE, deletes [1,2]
```

This seems much easier than backfilling the database. We could potentially turn this check back on after doing that, for now I just wanted those valid edits to stop failing to unbox for the sake of these tests.